### PR TITLE
modules: memfault: Implement location metrics

### PR DESCRIPTION
**Pull request by jtguggedal**
- Add location metrics in the Memfault SDK integration layer.
- Remove custom location metrics in Asset Tracker v2 and use the newly added ones instead.

### CODE REVIEW

**Comment by simensrostad** [comment]
:fireworks:

File: applications/asset_tracker_v2/doc/debug_module.rst [hunk 1]
```diff
@@ -25,13 +25,7 @@ This section documents the various features implemented by the module.
...
 ========
 
-The debug module uses `Memfault SDK`_ to track |NCS| specific metrics such as LTE and stack metrics.
-In addition, the following types of custom Memfault metrics are defined and tracked when compiling in the debug module:
-
- * ``gnss_time_to_fix_ms`` - Time duration between the start of a GNSS search and obtaining a fix.
- * ``gnss_satellites_tracked_count`` - Number of satellites tracked during a GNSS search window.
- * ``location_timeout_search_time_ms`` - Time duration between the start of a location search and a search timeout.
-
+The debug module uses `Memfault SDK`_ to track |NCS| specific metrics such as LTE, location, Bluetooth and stack metrics.
```
Comment (peknis) [inline applications/asset_tracker_v2/doc/debug_module.rst:28]:
```suggestion
The debug module uses `Memfault SDK`_ to track |NCS| specific metrics such as LTE, location, Bluetooth, and stack metrics.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -705,7 +705,10 @@ Common Application Framework
 Debug libraries
 ---------------
 
-|no_changes_yet_note|
+* :ref:`mod_memfault` library:
+
+  * Added location metrics, including GNSS, cellular and Wi-Fi specific metrics.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:710]:
```suggestion
  * Added location metrics, including GNSS, cellular, and Wi-Fi specific metrics.
```

**Review by peknis — APPROVED** [review]
Approved with a couple of minor nits.

**Review by trantanen — APPROVED** [review]

**Comment by jtguggedal** [comment]
CI passed, running only doc build after pushing only doc changes

**Pull request merged**